sdk/communication/communication-phone-numbers/src/sipRoutingClient.ts

```diff
@@ -0,0 +1,244 @@
+   */
+  public constructor(connectionString: string, options?: SipRoutingClientOptions);
+
+  /**
+   * Initializes a new instance of the SipRoutingClient class using an Azure KeyCredential.
+   *
+   * @param endpoint - The endpoint of the service (eg: https://contoso.eastus.communications.azure.net).
+   * @param credential - An object that is used to authenticate requests to the service. Use the Azure KeyCredential or `@azure/identity` to create a credential.
+   * @param options - Optional. Options to configure the HTTP pipeline.
+   */
+  public constructor(
+    endpoint: string,
+    credential: KeyCredential,
+    options?: SipRoutingClientOptions
+  );
+
+  /**
+   * Initializes a new instance of the SipRoutingClient class using a TokenCredential.
+   * @param endpoint - The endpoint of the service (ex: https://contoso.eastus.communications.azure.net).
+   * @param credential - TokenCredential that is used to authenticate requests to the service.
+   * @param options - Optional. Options to configure the HTTP pipeline.
+   */
+  public constructor(
+    endpoint: string,
+    credential: TokenCredential,
+    options?: SipRoutingClientOptions
+  );
+
+  public constructor(
+    connectionStringOrUrl: string,
+    credentialOrOptions?: KeyCredential | TokenCredential | SipRoutingClientOptions,
+    maybeOptions: SipRoutingClientOptions = {}
+  ) {
+    const { url, credential } = parseClientArguments(connectionStringOrUrl, credentialOrOptions);
+    const options = isSipClientOptions(credentialOrOptions) ? credentialOrOptions : maybeOptions;
+
+    const internalPipelineOptions: InternalPipelineOptions = {
+      ...options,
+      ...{
+        loggingOptions: {
+          logger: logger.info,
+        },
+      },
+    };
+
+    this.client = new SipRoutingGeneratedClient(url, {
+      endpoint: url,
+      ...internalPipelineOptions,
+    });
+    const authPolicy = createCommunicationAuthPolicy(credential);
+    this.client.pipeline.addPolicy(authPolicy);
+  }
+
+  /**
+   * Gets the SIP trunks.
+   * @param options - The options parameters.
+   */
+  public async getTrunks(options: OperationOptions = {}): Promise<SipTrunk[]> {
+    return tracingClient.withSpan("SipRoutingClient-getTrunks", options, async (updatedOptions) => {
+      const config = await this.client.getSipConfiguration(updatedOptions);
+      return transformFromRestModel(config.trunks);
+    });
+  }
+
+  /**
+   * Gets the SIP trunk.
+   * @param fqdn - The trunk's FQDN.
+   * @param options - The options parameters.
+   */
+  public async getTrunk(fqdn: string, options: OperationOptions = {}): Promise<SipTrunk> {
+    return tracingClient.withSpan("SipRoutingClient-getTrunk", options, async (updatedOptions) => {
+      const trunks = await this.getTrunks(updatedOptions);
+      const trunk = trunks.find((value: SipTrunk) => value.fqdn === fqdn);
+      if (trunk) {
+        return Promise.resolve(trunk);
```

Comment:
should be able to `return trunk;` since this is async arrow function?

Reply:
Refactored.